src/test/util_tests.cpp

```diff
@@ -550,57 +550,52 @@ BOOST_AUTO_TEST_CASE(util_ReadConfigStream)
-                && test_args.m_settings.ro_config[""].count("fff")
-                && test_args.m_settings.ro_config[""].count("ggg")
-                && test_args.m_settings.ro_config[""].count("h")
-                && test_args.m_settings.ro_config[""].count("i")
-               );
-    BOOST_CHECK(test_args.m_settings.ro_config["sec1"].count("ccc")
-                && test_args.m_settings.ro_config["sec1"].count("h")
-                && test_args.m_settings.ro_config["sec2"].count("ccc")
-                && test_args.m_settings.ro_config["sec2"].count("iii")
-               );
-
-    BOOST_CHECK(test_args.IsArgSet("-a")
-                && test_args.IsArgSet("-b")
-                && test_args.IsArgSet("-ccc")
-                && test_args.IsArgSet("-d")
-                && test_args.IsArgSet("-fff")
-                && test_args.IsArgSet("-ggg")
-                && test_args.IsArgSet("-h")
-                && test_args.IsArgSet("-i")
-                && !test_args.IsArgSet("-zzz")
-                && !test_args.IsArgSet("-iii")
-               );
-
-    BOOST_CHECK(test_args.GetArg("-a", "xxx") == ""
-                && test_args.GetArg("-b", "xxx") == "1"
-                && test_args.GetArg("-ccc", "xxx") == "argument"
-                && test_args.GetArg("-d", "xxx") == "e"
-                && test_args.GetArg("-fff", "xxx") == "0"
-                && test_args.GetArg("-ggg", "xxx") == "1"
-                && test_args.GetArg("-h", "xxx") == "0"
-                && test_args.GetArg("-i", "xxx") == "1"
-                && test_args.GetArg("-zzz", "xxx") == "xxx"
-                && test_args.GetArg("-iii", "xxx") == "xxx"
-               );
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("a"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("b"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("ccc"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("d"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("fff"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("ggg"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("h"));
+    BOOST_CHECK(test_args.m_settings.ro_config[""].count("i"));
+    BOOST_CHECK(test_args.m_settings.ro_config["sec1"].count("ccc"));
+    BOOST_CHECK(test_args.m_settings.ro_config["sec1"].count("h"));
+    BOOST_CHECK(test_args.m_settings.ro_config["sec2"].count("ccc"));
+    BOOST_CHECK(test_args.m_settings.ro_config["sec2"].count("iii"));
+
+    BOOST_CHECK(test_args.IsArgSet("-a"));
+    BOOST_CHECK(test_args.IsArgSet("-b"));
+    BOOST_CHECK(test_args.IsArgSet("-ccc"));
+    BOOST_CHECK(test_args.IsArgSet("-d"));
+    BOOST_CHECK(test_args.IsArgSet("-fff"));
+    BOOST_CHECK(test_args.IsArgSet("-ggg"));
+    BOOST_CHECK(test_args.IsArgSet("-h"));
+    BOOST_CHECK(test_args.IsArgSet("-i"));
+    BOOST_CHECK(!test_args.IsArgSet("-zzz"));
+    BOOST_CHECK(!test_args.IsArgSet("-iii"));
+
+    BOOST_CHECK(test_args.GetArg("-a", "xxx") == "");
```

Comment:
```suggestion
    BOOST_CHECK_EQUAL(test_args.GetArg("-a", "xxx"), "");
```

would produce a better error message, if it fails some day.

Same in all places below where `BOOST_CHECK(a == b);` can be replaced with `BOOST_CHECK_EQUAL(a, b);`

Reply:
Thank you! [Updated](https://github.com/bitcoin/bitcoin/pull/19709#issuecomment-673983005).